Fix `test_hermes`, which was not being run properly due to #7869

### Description

See https://github.com/rucio/rucio/issues/7869 for reference

https://github.com/rucio/rucio/actions/runs/16370632700/job/46257897936?pr=7871

```
=================================== FAILURES ===================================
_ test_hermes[caches_mock0-core_config_mock{'hermes': {'services_list': 'influx,activemq,elastic,email', 'elastic_endpoint': 'http://elasticsearch:9200/ddm_events/doc/_bulk', 'influxdb_endpoint': 'http://influxdb:8086/api/v2/write?org=rucio&bucket=rucio', 'influxdb_token': 'mytoken'}, 'messaging-hermes': {'destination': '/queue/events', 'brokers': 'localhost', 'use_ssl': False, 'username': 'hermes', 'password': 'supersecret', 'nonssl_port': 61613, 'send_email': False, 'smtp-host': 'testing.host', 'smtp-port': 1234}}] _
tests/test_hermes.py:173: in test_hermes
    smtp_mock.assert_called_with(host="testing.host", port=1234)
/usr/lib64/python3.9/unittest/mock.py:898: in assert_called_with
    raise AssertionError(error_message)
E   AssertionError: expected call not found.
E   Expected: mock(host='testing.host', port=1234)
E   Actual: not called.
```

We should merge https://github.com/rucio/rucio/pull/7868 first, then make a PR to fix the test, and then merge https://github.com/rucio/rucio/pull/7871


### Steps to reproduce

Run CI

### Rucio Version

_No response_

### Additional Information

_No response_

--- a/etc/docker/test/matrix.yml
+++ b/etc/docker/test/matrix.yml
@@ -37,8 +37,6 @@ suites:
       - oracle
       - mysql8
       - postgres14
-    services:
-      - influxdb_elastic
   - id: sqlite
     RDBMS:
       - sqlite

--- a/etc/docker/test/matrix_nightly.yml
+++ b/etc/docker/test/matrix_nightly.yml
@@ -19,8 +19,6 @@ suites:
       - oracle
       - mysql8
       - postgres14
-    services:
-      - influxdb_elastic
   - id: sqlite
     RDBMS:
       - sqlite

--- a/tests/test_hermes.py
+++ b/tests/test_hermes.py
@@ -76,7 +76,9 @@ class MyListener:
                 ("messaging-hermes", "username", "hermes"),
                 ("messaging-hermes", "password", "supersecret"),
                 ("messaging-hermes", "nonssl_port", 61613),
-                ("messaging-hermes", "send_email", False),
+                ("messaging-hermes", "send_email", True),
+                ("messaging-hermes", "smtp_host", "testing.host"),
+                ("messaging-hermes", "smtp_port", 1234),
             ]
         },
         {
@@ -99,9 +101,9 @@ class MyListener:
                 ("messaging-hermes", "username", "hermes"),
                 ("messaging-hermes", "password", "supersecret"),
                 ("messaging-hermes", "nonssl_port", 61613),
-                ("messaging-hermes", "send_email", False),
-                ("messaging-hermes", "smtp-host", "testing.host"),
-                ("messaging-hermes", "smtp-port", 1234),
+                ("messaging-hermes", "send_email", True),
+                ("messaging-hermes", "smtp_host", "testing.host"),
+                ("messaging-hermes", "smtp_port", 1234),
             ]
         }
     ],
@@ -205,7 +207,11 @@ def test_hermes(core_config_mock, caches_mock, monkeypatch):
         smtp_mock = MagicMock()
         m.setattr(hermes.smtplib, "SMTP", smtp_mock)
         hermes.hermes(once=True)
-        smtp_mock.assert_called_with(host="testing.host", port=1234)
+        smtp_host = config_get("messaging-hermes", "smtp_host", default='', raise_exception=False)
+        if not smtp_host:
+            smtp_mock.assert_called_with()
+        else:
+            smtp_mock.assert_called_with(host="testing.host", port=1234)
     service_dict = {"influx": 0, "elastic": 0, "email": 0, "activemq": 0}
     messages = retrieve_messages(50, old_mode=False)
     for message in messages:

